tests/returntypes/test_tensorflow_qnode_shot_vector.py

```diff
@@ -0,0 +1,581 @@
+# Copyright 2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Integration tests for using the TF interface with shot vectors and with a QNode"""
+
+import pytest
+
+import pennylane as qml
+from pennylane import numpy as np
+from pennylane import qnode
+
+pytestmark = pytest.mark.tf
+
+tf = pytest.importorskip("tensorflow")
+
+shots = [((5, 2), 1, 10)]
+
+qubit_device_and_diff_method = [
+    ["default.qubit", "finite-diff", {"h": 10e-2}],
+    ["default.qubit", "parameter-shift", {}],
+]
+
+
+@pytest.mark.parametrize("shots", shots)
+@pytest.mark.parametrize("dev_name,diff_method,gradient_kwargs", qubit_device_and_diff_method)
+class TestReturnWithShotVectors:
+    """Class to test the shape of the Grad/Jacobian/Hessian with different return types and shot vectors."""
+
+    def test_jac_single_measurement_param(self, dev_name, diff_method, gradient_kwargs, shots):
+        """For one measurement and one param, the gradient is a float."""
+        dev = qml.device(dev_name, wires=1, shots=shots)
+
+        @qnode(dev, diff_method=diff_method, interface="tf", **gradient_kwargs)
+        def circuit(a):
+            qml.RY(a, wires=0)
+            qml.RX(0.2, wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+        a = tf.Variable(0.1)
+
+        with tf.GradientTape() as tape:
+            res = circuit(a)
+            res = qml.math.stack(res)
+
+        jac = tape.jacobian(res, a)
+
+        assert isinstance(jac, tf.Tensor)
+        num_copies = sum(
+            [1 for x in shots if isinstance(x, int)] + [x[1] for x in shots if isinstance(x, tuple)]
+        )
```

Comment:
It would be more readable for the tests if rather than calculating the number of copies programmatically, we'd have them hard coded and passed into each test case. (Just realized this, Autograd tests had similar statements.)